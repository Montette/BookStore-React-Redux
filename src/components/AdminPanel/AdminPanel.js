import React from 'react';
import './AdminPanel.scss';
import {fbase, firebaseApp} from '../../fbase';
const uuidv1 = require('uuid/v1');


class AdminPanel extends React.Component {
    state = {
        books: [],
      book: {
          name: '',
          author: '',
          description: '',
          image: '',
          onStock: false
      },
      loggedIn: false,
      email: '',
      password:''
    }

    inputHandler = (event) => {

        let newVal = event.target.value;
        let updatedState = {};
        if(event.target.type === 'checkbox') {
            updatedState[event.target.name] = event.target.checked;
        } else {
            updatedState[event.target.name] = newVal;
        }
  
        this.setState({
            book: {
                ...this.state.book,
                ...updatedState
            }
        })
  
    }

    submitHandler = (event) => {
        event.preventDefault();
        // let books = [...this.state.books];
        const book = {...this.state.book};
        book.id = uuidv1();
        // books.push(book); 
        let resetBook = {...this.state.book};
        // this.props.addBook(book)
        Object.keys(resetBook).map(key=> {
              resetBook[key] = typeof resetBook[key] === 'boolean' ? false : '';
        });
        this.setState({
            books: [...this.state.books, book],
            book: resetBook    
        }) 
    }

    componentDidMount () {
        this.ref = fbase.syncState('bookstore/books', {
            context: this,
            state: 'books'
        })
    }

    componentWillUnmount (){
        fbase.removeBinding(this.ref)
    }

    authenticateHandler = (event)=> {
        event.preventDefault();
        firebaseApp.auth().signInWithEmailAndPassword(this.state.email, this.state.password)
            .then(()=> {
                this.setState({
                    loggedIn: true
                })
            })
            .catch(()=> {
                console.log('wrong email or password')
            })
    }

    loginHandleChange = (event)=> {
        this.setState({
            [event.target.name]: event.target.value
        })
    }
    
    render(){
        
        return (
            <React.Fragment>
            {!this.state.loggedIn &&
            <form onSubmit={this.authenticateHandler}>
            <input type='text' placeholder='email' name='email'
            onChange={this.loginHandleChange}
            />
               <input type='password' name='password'
            onChange={this.loginHandleChange}
            />
            <button type='submit'>Log in</button>
            </form>
            }
            {this.state.loggedIn &&
            <div className='adminPanel'>
                <form onSubmit={this.submitHandler} className='adminPanel__form form'>
                    
                    <label className='form__label' htmlFor='bookNameInput'>Book name:
                    <input className='form__input' type='text' placeholder='Book Name' id='bookName' name='name' value={this.state.book.name} onChange={this.inputHandler}/>
                    </label>
                    <label className='form__label' htmlFor='bookAuthorInput'>Book author:
                    <input className='form__input' type='text' placeholder='Book Author' id='bookAuthor' name='author' value={this.state.book.author} onChange={this.inputHandler}/>
                    </label>
                    <label className='form__label' htmlFor='bookAuthorInput'>Book description:
                    <textarea className='form__textarea'  placeholder='Book Description' id='bookDescription' name='description' value={this.state.book.description} onChange={this.inputHandler}/>
                    </label>
                    <label className='form__label' htmlFor='bookAuthorInput'>Book Image:
                    <input className='form__input' type='file' id='bookImage' name='image' value={this.state.book.image} onChange={this.inputHandler}/>
                    </label>
                    <label className='form__label' htmlFor='bookAuthorInput'>On stock:
                    <input className='form__input' type='checkbox'  placeholder='bookOnStock' id='bookOnStock' name='onStock' value={this.state.book.name} onChange={this.inputHandler}/>
                    </label>
                    <button type='submite'>Save</button>
                </form>
            
            </div>
            }
            </React.Fragment>
        );
    }
}

export default AdminPanel;