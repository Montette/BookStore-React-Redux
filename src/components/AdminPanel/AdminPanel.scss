.form {
    font-size: 13px;
    padding: 8px 10px;
    display: flex;
    max-width: 400px;
    width: 100%;
    flex-wrap: wrap;
    text-align: left;
    &__input {
        display: block;
        padding-left: 5px;
        color: darkGrey;
        background-color: #fdfcfc;
        margin-top: 5px;
        height: 30px;
        border-radius: 3px;
        width: 100%;
    }
    &__label {
        display: block;
        margin-bottom: 20px;
        margin-top: 7px;
        font-weight: bold;
        color: darkGrey;
        width: 100%;
        
        
    }
    &__textarea{
        display: block;
        margin-bottom: 20px;
        margin-top: 7px;
        font-weight: bold;
        color: darkGrey;
        width: 100%;
    }


}



