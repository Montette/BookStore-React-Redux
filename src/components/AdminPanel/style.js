import styled from 'styled-components';

export const Form = styled.form`
font-size: 13px;
padding: 8px 10px;
display: flex;
max-width: 400px;
width: 100%;
flex-wrap: wrap;
text-align: left;

`

export const Input = styled.input`
display: block;
padding-left: 5px;
color: darkGrey;
background-color: #fdfcfc;
margin-top: 5px;
height: 30px;
border-radius: 3px;
width: 100%;
`
export const Label = styled.label`
        display: block;
        margin-bottom: 20px;
        margin-top: 7px;
        font-weight: bold;
        color: darkGrey;
        width: 100%;
`
export const TextArea = styled.textarea`
display: block;
margin-bottom: 20px;
margin-top: 7px;
font-weight: bold;
color: darkGrey;
width: 100%;
`