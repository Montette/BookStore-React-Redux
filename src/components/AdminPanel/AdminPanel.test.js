import React from 'react';
import ReactDOM from 'react-dom';
import AdminPanel from './AdminPanel';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure ({adapter: new Adapter()})

describe('AdminPanel tests', ()=> {


    it('AdminPanel renders without crashing', ()=> {
       shallow(<AdminPanel />)    
    })

    
    it('Snapshot matches', ()=> {
        const wrapper = shallow(<AdminPanel />);
        expect(wrapper).toMatchSnapshot();

     })

})