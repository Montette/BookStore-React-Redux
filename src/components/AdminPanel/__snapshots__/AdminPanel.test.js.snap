// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPanel tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="form"
>
        <label
                className="form__label"
                htmlFor="bookNameInput"
        >
                Book name:
                <input
                        className="form__input"
                        id="bookName"
                        name="bookName"
                        placeholder="Book Name"
                        type="text"
                />
        </label>
        <label
                className="form__label"
                htmlFor="bookAuthorInput"
        >
                Book author:
                <input
                        className="form__input"
                        id="bookAuthor"
                        name="bookAuthor"
                        placeholder="Book Author"
                        type="text"
                />
        </label>
        <label
                className="form__label"
                htmlFor="bookAuthorInput"
        >
                Book description:
                <textarea
                        className="form__textarea"
                        id="bookDescription"
                        name="bookDescription"
                        placeholder="Book Description"
                />
        </label>
        <label
                className="form__label"
                htmlFor="bookAuthorInput"
        >
                Book Image:
                <input
                        className="form__input"
                        id="bookImage"
                        name="bookImage"
                        type="file"
                />
        </label>
        <label
                className="form__label"
                htmlFor="bookAuthorInput"
        >
                On stock:
                <input
                        className="form__input"
                        id="bookOnStock"
                        name="bookOnStock"
                        placeholder="bookOnStock"
                        type="checkbox"
                />
        </label>
        <button
                type="submite"
        >
                Save
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="form__label"
            htmlFor="bookNameInput"
>
            Book name:
            <input
                        className="form__input"
                        id="bookName"
                        name="bookName"
                        placeholder="Book Name"
                        type="text"
            />
</label>,
          <label
            className="form__label"
            htmlFor="bookAuthorInput"
>
            Book author:
            <input
                        className="form__input"
                        id="bookAuthor"
                        name="bookAuthor"
                        placeholder="Book Author"
                        type="text"
            />
</label>,
          <label
            className="form__label"
            htmlFor="bookAuthorInput"
>
            Book description:
            <textarea
                        className="form__textarea"
                        id="bookDescription"
                        name="bookDescription"
                        placeholder="Book Description"
            />
</label>,
          <label
            className="form__label"
            htmlFor="bookAuthorInput"
>
            Book Image:
            <input
                        className="form__input"
                        id="bookImage"
                        name="bookImage"
                        type="file"
            />
</label>,
          <label
            className="form__label"
            htmlFor="bookAuthorInput"
>
            On stock:
            <input
                        className="form__input"
                        id="bookOnStock"
                        name="bookOnStock"
                        placeholder="bookOnStock"
                        type="checkbox"
            />
</label>,
          <button
            type="submite"
>
            Save
</button>,
        ],
        "className": "form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Book name:",
              <input
                className="form__input"
                id="bookName"
                name="bookName"
                placeholder="Book Name"
                type="text"
/>,
            ],
            "className": "form__label",
            "htmlFor": "bookNameInput",
          },
          "ref": null,
          "rendered": Array [
            "Book name:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "id": "bookName",
                "name": "bookName",
                "placeholder": "Book Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Book author:",
              <input
                className="form__input"
                id="bookAuthor"
                name="bookAuthor"
                placeholder="Book Author"
                type="text"
/>,
            ],
            "className": "form__label",
            "htmlFor": "bookAuthorInput",
          },
          "ref": null,
          "rendered": Array [
            "Book author:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "id": "bookAuthor",
                "name": "bookAuthor",
                "placeholder": "Book Author",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Book description:",
              <textarea
                className="form__textarea"
                id="bookDescription"
                name="bookDescription"
                placeholder="Book Description"
/>,
            ],
            "className": "form__label",
            "htmlFor": "bookAuthorInput",
          },
          "ref": null,
          "rendered": Array [
            "Book description:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__textarea",
                "id": "bookDescription",
                "name": "bookDescription",
                "placeholder": "Book Description",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Book Image:",
              <input
                className="form__input"
                id="bookImage"
                name="bookImage"
                type="file"
/>,
            ],
            "className": "form__label",
            "htmlFor": "bookAuthorInput",
          },
          "ref": null,
          "rendered": Array [
            "Book Image:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "id": "bookImage",
                "name": "bookImage",
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "On stock:",
              <input
                className="form__input"
                id="bookOnStock"
                name="bookOnStock"
                placeholder="bookOnStock"
                type="checkbox"
/>,
            ],
            "className": "form__label",
            "htmlFor": "bookAuthorInput",
          },
          "ref": null,
          "rendered": Array [
            "On stock:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form__input",
                "id": "bookOnStock",
                "name": "bookOnStock",
                "placeholder": "bookOnStock",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "type": "submite",
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="form"
>
          <label
                    className="form__label"
                    htmlFor="bookNameInput"
          >
                    Book name:
                    <input
                              className="form__input"
                              id="bookName"
                              name="bookName"
                              placeholder="Book Name"
                              type="text"
                    />
          </label>
          <label
                    className="form__label"
                    htmlFor="bookAuthorInput"
          >
                    Book author:
                    <input
                              className="form__input"
                              id="bookAuthor"
                              name="bookAuthor"
                              placeholder="Book Author"
                              type="text"
                    />
          </label>
          <label
                    className="form__label"
                    htmlFor="bookAuthorInput"
          >
                    Book description:
                    <textarea
                              className="form__textarea"
                              id="bookDescription"
                              name="bookDescription"
                              placeholder="Book Description"
                    />
          </label>
          <label
                    className="form__label"
                    htmlFor="bookAuthorInput"
          >
                    Book Image:
                    <input
                              className="form__input"
                              id="bookImage"
                              name="bookImage"
                              type="file"
                    />
          </label>
          <label
                    className="form__label"
                    htmlFor="bookAuthorInput"
          >
                    On stock:
                    <input
                              className="form__input"
                              id="bookOnStock"
                              name="bookOnStock"
                              placeholder="bookOnStock"
                              type="checkbox"
                    />
          </label>
          <button
                    type="submite"
          >
                    Save
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="form__label"
              htmlFor="bookNameInput"
>
              Book name:
              <input
                            className="form__input"
                            id="bookName"
                            name="bookName"
                            placeholder="Book Name"
                            type="text"
              />
</label>,
            <label
              className="form__label"
              htmlFor="bookAuthorInput"
>
              Book author:
              <input
                            className="form__input"
                            id="bookAuthor"
                            name="bookAuthor"
                            placeholder="Book Author"
                            type="text"
              />
</label>,
            <label
              className="form__label"
              htmlFor="bookAuthorInput"
>
              Book description:
              <textarea
                            className="form__textarea"
                            id="bookDescription"
                            name="bookDescription"
                            placeholder="Book Description"
              />
</label>,
            <label
              className="form__label"
              htmlFor="bookAuthorInput"
>
              Book Image:
              <input
                            className="form__input"
                            id="bookImage"
                            name="bookImage"
                            type="file"
              />
</label>,
            <label
              className="form__label"
              htmlFor="bookAuthorInput"
>
              On stock:
              <input
                            className="form__input"
                            id="bookOnStock"
                            name="bookOnStock"
                            placeholder="bookOnStock"
                            type="checkbox"
              />
</label>,
            <button
              type="submite"
>
              Save
</button>,
          ],
          "className": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Book name:",
                <input
                  className="form__input"
                  id="bookName"
                  name="bookName"
                  placeholder="Book Name"
                  type="text"
/>,
              ],
              "className": "form__label",
              "htmlFor": "bookNameInput",
            },
            "ref": null,
            "rendered": Array [
              "Book name:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form__input",
                  "id": "bookName",
                  "name": "bookName",
                  "placeholder": "Book Name",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Book author:",
                <input
                  className="form__input"
                  id="bookAuthor"
                  name="bookAuthor"
                  placeholder="Book Author"
                  type="text"
/>,
              ],
              "className": "form__label",
              "htmlFor": "bookAuthorInput",
            },
            "ref": null,
            "rendered": Array [
              "Book author:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form__input",
                  "id": "bookAuthor",
                  "name": "bookAuthor",
                  "placeholder": "Book Author",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Book description:",
                <textarea
                  className="form__textarea"
                  id="bookDescription"
                  name="bookDescription"
                  placeholder="Book Description"
/>,
              ],
              "className": "form__label",
              "htmlFor": "bookAuthorInput",
            },
            "ref": null,
            "rendered": Array [
              "Book description:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form__textarea",
                  "id": "bookDescription",
                  "name": "bookDescription",
                  "placeholder": "Book Description",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Book Image:",
                <input
                  className="form__input"
                  id="bookImage"
                  name="bookImage"
                  type="file"
/>,
              ],
              "className": "form__label",
              "htmlFor": "bookAuthorInput",
            },
            "ref": null,
            "rendered": Array [
              "Book Image:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form__input",
                  "id": "bookImage",
                  "name": "bookImage",
                  "type": "file",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "On stock:",
                <input
                  className="form__input"
                  id="bookOnStock"
                  name="bookOnStock"
                  placeholder="bookOnStock"
                  type="checkbox"
/>,
              ],
              "className": "form__label",
              "htmlFor": "bookAuthorInput",
            },
            "ref": null,
            "rendered": Array [
              "On stock:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form__input",
                  "id": "bookOnStock",
                  "name": "bookOnStock",
                  "placeholder": "bookOnStock",
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save",
              "type": "submite",
            },
            "ref": null,
            "rendered": "Save",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
